sql/src/test/java/io/crate/operation/operator/InOperatorTest.java

```diff
@@ -20,175 +20,56 @@
  */
 package io.crate.operation.operator;
 
-import com.google.common.collect.Sets;
-import io.crate.action.sql.SessionContext;
-import io.crate.analyze.symbol.Function;
 import io.crate.analyze.symbol.Literal;
-import io.crate.analyze.symbol.Symbol;
-import io.crate.metadata.Reference;
-import io.crate.metadata.TransactionContext;
-import io.crate.operation.operator.input.ObjectInput;
-import io.crate.test.integration.CrateUnitTest;
-import io.crate.types.DataType;
+import io.crate.operation.operator.any.AnyOperator;
+import io.crate.operation.scalar.AbstractScalarFunctionsTest;
 import io.crate.types.DataTypes;
-import io.crate.types.SetType;
-import org.apache.lucene.util.BytesRef;
 import org.junit.Test;
 
-import java.util.ArrayList;
-import java.util.HashSet;
-import java.util.List;
-import java.util.Set;
-
+import static io.crate.testing.SymbolMatchers.isFunction;
 import static io.crate.testing.SymbolMatchers.isLiteral;
-import static org.hamcrest.Matchers.instanceOf;
-import static org.hamcrest.Matchers.is;
-
-public class InOperatorTest extends CrateUnitTest {
 
-    private final TransactionContext transactionContext = new TransactionContext(SessionContext.SYSTEM_SESSION);
-
-    private static final DataType INTEGER_SET_TYPE = new SetType(DataTypes.INTEGER);
-    private static final DataType STRING_SET_TYPE = new SetType(DataTypes.STRING);
+public class InOperatorTest extends AbstractScalarFunctionsTest {
 
     @Test
     public void testNormalizeSymbolSetLiteralIntegerIncluded() {
-        Literal<Integer> inValue = Literal.of(1);
-        Literal inListValues = Literal.of(INTEGER_SET_TYPE, Sets.newHashSet(1, 2, 4, 8));
-
-        List<Symbol> arguments = new ArrayList<>();
-        arguments.add(inValue);
-        arguments.add(inListValues);
-
-        InOperator op = new InOperator(Operator.generateInfo(InOperator.NAME, DataTypes.INTEGER));
-        Function function = new Function(op.info(), arguments);
-        Symbol result = op.normalizeSymbol(function, transactionContext);
-
-        assertThat(result, isLiteral(true));
+        assertNormalize("1 in (1, 2, 4, 8)", isLiteral(true));
     }
 
     @Test
     public void testNormalizeSymbolSetLiteralIntegerNotIncluded() {
-        Literal<Integer> inValue = Literal.of(128);
-
-        Literal inListValues = Literal.of(INTEGER_SET_TYPE, Sets.newHashSet(1, 2, 4, 8));
-
-        List<Symbol> arguments = new ArrayList<>();
-        arguments.add(inValue);
-        arguments.add(inListValues);
-
-        InOperator op = new InOperator(Operator.generateInfo(InOperator.NAME, DataTypes.INTEGER));
-        Function function = new Function(op.info(), arguments);
-        Symbol result = op.normalizeSymbol(function, transactionContext);
-
-        assertThat(result, isLiteral(false));
+        assertNormalize("128 in (1, 2, 4, 8)", isLiteral(false));
```

Comment:
I'd like to keep it in separate tests, it's usually easiser to debug if e.g. the third statement has an issue.
In the other tests I mostly didn't split it up because of laziness =)